src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp

```diff
@@ -66,7 +66,43 @@ void SunSearch::Reset(uint64_t CurrentSimNanos)
  */
 void SunSearch::UpdateState(uint64_t CurrentSimNanos)
 {
+    /*! create and zero the output message */
+    AttGuidMsgPayload attGuidOut = this->attGuidOutMsg.zeroMsgPayload;
 
+    /*! read vehicle configuration message */
+    NavAttMsgPayload attNavIn = this->attNavInMsg();
+
+    double omega_BN_B[3] = {attNavIn.omega_BN_B[0], attNavIn.omega_BN_B[1], attNavIn.omega_BN_B[2]};
+
+    double omega_RN_B[3] = {0, 0, 0};
+    double domega_RN_B[3] = {0, 0, 0};
+
+    double CurrentSimSeconds = (CurrentSimNanos - this->resetTime) * NANO2SEC;
+
+    double timeInf = 0;
+    double timeSup = this->slewTotalTime[0];
+    for (int index=0; index<3; ++index) {
+        if (CurrentSimSeconds >= timeInf && CurrentSimSeconds < timeSup) {
+            ComputeReferenceMotion(CurrentSimNanos, index, &omega_RN_B[0], &domega_RN_B[0]);
```

Comment:
:large_orange_diamond: this->computeRef...

Reply:
Done